backend/infrahub/core/diff/conflicts_identifier.py

```diff
@@ -0,0 +1,266 @@
+            conflict_path=path,
+            path_type=PathType.NODE,
+            property_name=None,
+            changes=branch_changes,
+        )
+
+    def _build_attribute_conflicts(
+        self,
+        branch_node: EnrichedDiffNode,
+        base_attribute: EnrichedDiffAttribute,
+        branch_attribute: EnrichedDiffAttribute,
+    ) -> list[DataConflict]:
+        conflicts = []
+        base_property_map = {p.property_type: p for p in base_attribute.properties}
+        branch_property_map = {p.property_type: p for p in branch_attribute.properties}
+        common_property_types = set(base_property_map.keys()) & set(branch_property_map.keys())
+        for property_type in common_property_types:
+            base_property = base_property_map[property_type]
+            branch_property = branch_property_map[property_type]
+            if base_property.new_value != branch_property.new_value:
+                conflict = self._add_property_conflict(
+                    branch_node=branch_node,
+                    branch_attribute=branch_attribute,
+                    base_property=base_property,
+                    branch_property=branch_property,
+                )
+                conflicts.append(conflict)
+        return conflicts
+
+    def _build_relationship_conflicts(
+        self,
+        branch_node: EnrichedDiffNode,
+        base_relationship: EnrichedDiffRelationship,
+        branch_relationship: EnrichedDiffRelationship,
+    ) -> list[DataConflict]:
+        conflicts: list[DataConflict] = []
+        node_schema = self.schema_manager.get_node_schema(
+            name=branch_node.kind, branch=self.diff_branch, duplicate=False
+        )
+        relationship_schema = node_schema.get_relationship(name=branch_relationship.name)
+        is_cardinality_one = relationship_schema.cardinality is RelationshipCardinality.ONE
+        if is_cardinality_one:
+            base_element = next(iter(base_relationship.relationships))
+            branch_element = next(iter(branch_relationship.relationships))
+            element_tuples = [(base_element, branch_element)]
+        else:
+            base_peer_id_map = {element.peer_id: element for element in base_relationship.relationships}
+            branch_peer_id_map = {element.peer_id: element for element in branch_relationship.relationships}
+            common_peer_ids = set(base_peer_id_map.keys()) & set(branch_peer_id_map.keys())
+            element_tuples = [(base_peer_id_map[peer_id], branch_peer_id_map[peer_id]) for peer_id in common_peer_ids]
+        for base_element, branch_element in element_tuples:
+            conflicts.extend(
+                self._add_relationship_conflicts_for_one_peer(
+                    branch_node=branch_node,
+                    branch_relationship=branch_relationship,
+                    base_element=base_element,
+                    branch_element=branch_element,
+                    is_cardinality_one=is_cardinality_one,
```

Comment:
How does this work, if we add a relationship to the same peer it gets marked as a conflict?

Reply:
not quite, if the same property is changed on a relationship for a given peer ID, that would be a conflict